fakeldap.py

```diff
@@ -440,22 +449,29 @@ def _add_s(self, dn, record):
             self.directory[dn] = entry
             return (105,[], len(self.calls), [])
 
-    def _simple_onelevel_search(self, base, filterstr):
-        search_attr_name, search_attr_value = filterstr[1:-1].split('=')
-
+    def _multiple_attrs_onelevel_search(self, base, attrs_conditions):
```

Comment:
Would be helpful to add a docstring here that `attrs_conditions` are considered to be `AND`-joined because that's the only operation we support

Reply:
see 36297324ff0bd2128646333525260cde568d2b16

diff:
```diffdiff --git a/fakeldap.py b/fakeldap.py
index 03e8437..c87d49c 100644
--- a/fakeldap.py
+++ b/fakeldap.py
@@ -450,6 +450,12 @@ class MockLDAP(object):
             return (105,[], len(self.calls), [])
 
     def _multiple_attrs_onelevel_search(self, base, attrs_conditions):
+        """Search the base with tuples of conditions that are "anded" together
+        (all conditions have to be met)
+
+        attrs_conditions -- List of tuples, first element of each tuple is the attribute
+                            name. Second element is the required value.
+        """
         result = []
         for dn, attrs in self.directory.items():
             if dn.endswith(',{}'.format(base)):
```

Reply:
ah, I got it and added option to match list at the same time:

308ecb10570d15d41c28d0bddfdfd0681e437dac

```python

                match = True
                for search_attr_name, search_attr_value in attrs_conditions:
                    search_attr = attrs.get(search_attr_name)
                    if search_attr == search_attr_value:
                        continue
                    elif isinstance(search_attr, list) and search_attr_value in search_attr:  # if attr is in the format "attr_name": ["value", ]
                        continue
                    else:
                        match = False
                        break

                if match:
                    result.append((dn, attrs))
```

Reply:
Sounds good to me